### ISSUE TITLE
eXist-db ignores password for guest user

### ISSUE THREAD

**Issue by lguariento** [eXist-db/exist#1781]
### What is the problem
> The browser access as guest an app which is restricted to a custom group without asking for user/password, even if a password for guest has been set and guest is member of the custom group.

### What did you expect
> I expect the browser to ask for user/password as it does if the guest user is not member of the custom group.

### Describe how to reproduce or add a test
> Create a .html file within your app which has a read-write-execute enabled custom group (say, mygroup) in its access control list, and for which 'Other' has no permissions. Assign guest to mygroup. Give 'guest' a password. Access that html file on your browser. The browser will load that file (as guest) without asking for any authentication.

### Context information

- eXist-db 4.0.0
- Java 1.8.0_152-b16
- CentOS 7
- 64 bit

**Comment by joewiz** [comment]
Does your app have a controller.xql? What is the directive I’m in operation for the HTML file in question?

**Comment by lguariento** [comment]
Yes, I have a controller.xql for my application:

```
xquery version "3.0";

declare variable $exist:path external;
declare variable $exist:resource external;
declare variable $exist:controller external;
declare variable $exist:prefix external;
declare variable $exist:root external;

if ($exist:path eq '') then
    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
        <redirect url="{request:get-uri()}/"/>
    </dispatch>
    
else if ($exist:path eq "/") then
    (: forward root path to index.xql :)
    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
        <redirect url="pages/index.html"/>
    </dispatch>
    
else if (ends-with($exist:resource, ".html")) then
    (: the html page is run through view.xql to expand templates :)
    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
        <view>
            <forward url="{$exist:controller}/modules/view.xql"/>
        </view>
		<error-handler>
			<forward url="{$exist:controller}/error-page.html" method="get"/>
			<forward url="{$exist:controller}/modules/view.xql"/>
		</error-handler>
    </dispatch>
    
(: Resource paths starting with $app-root are loaded from the application's root collection :)
else if (contains($exist:path,"app-root")) then
<dispatch xmlns="http://exist.sourceforge.net/NS/exist">
    <forward url="{$exist:controller}/{substring-after($exist:path, '$app-root/')}">
        <set-header name='Cache-Control' value="no"/>
    </forward>
</dispatch>
    
(: Resource paths starting with $shared are loaded from the shared-resources app :)
else if (contains($exist:path, "/$shared/")) then
    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
        <forward url="/shared-resources/{substring-after($exist:path, '/$shared/')}">
            <set-header name="Cache-Control" value="max-age=3600, must-revalidate"/>
        </forward>
    </dispatch>
else
    (: everything else is passed through :)
    
    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
        <cache-control cache="yes"/>
    </dispatch>

```

The files are in myapp/pages/*.html .

**Comment by line-o** [comment]
@lguariento You said this is an `app which is restricted to a custom group`. How did you restrict access? 
The controller imposes no restrictions / defines all resources as public. In my experience this is the only reliable way to enforce access restrictions for HTTP requests.

**Comment by lguariento** [comment]
The application does precisely what I want with my custom group. I created my user and the group via the java admin client, put my users inside the group, and added an ACL for the resources (e.g. pages/something.html) I want to restrict. When the users access those resources (e.g. pages/something.html) the browser will ask for authentication, which is what happens now.

Is there another way of doing that that I'm not aware? I posted my controller only upon @joewiz 's request - can the controller restrict access to certain users?
